### ISSUE TITLE
Old style key signatures can't be placed at the head of a line

### ISSUE THREAD

**Issue by RhinoHaggis** [musescore/MuseScore#24922]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Description with steps to reproduce

Current behavior:
With old key signatures  (naturals cancel the previous key signature), the naturals cannot be displayed at the head of a line:

1. Add several blank measures
2. Style > Accidentals > "natural in key signature: Before key signature if changing to fewer sharps or flats"
3. Add key signature changes that require the cancelling natural signs.
4.  Try to get a new key signature _with the cancelling naturals_ at the head of the line.

Expected behavior:
Absent any obstacle such as section breaks, the complete key signature should be displayed.

### Supporting files, videos and screenshots

Sample file with blank measures and two key signatures
[old man key signatures.mscz.zip](https://github.com/user-attachments/files/17138266/old.man.key.signatures.mscz.zip)

The goal is to obtain this:
![key sig at head of line](https://github.com/user-attachments/assets/73c82d1f-e433-43f4-a1b6-35d7eeaa9cdf)
with one of the key signatures.

If, by chance, you succeed, try to replicate your success.

### What is the latest version of MuseScore Studio where this issue is present?

4.4

### Regression

No.

### Operating system

MacOS 12.6

### Additional context

The issues are present in 4.2 and also 3.6.
The behavior seems inappropriately tied to "show courtesy key signature on previous system".  

- For a signature at the head of a line (not necessarily after a line break), the naturals always display on a courtesy signature, but turning off "courtesy" eliminates them from the signature altogether.

- Signatures in the middle of a line are more stable, and usually always show the naturals.  However, toggling "show courtesy" may -or may not- toggle the naturals.

The behavior is also unpredictable.  These two signatures:
![lucky break](https://github.com/user-attachments/assets/dd95bac9-1341-4c21-9991-c1fe3f6ce86b)
both have courtesy accidentals turned off.  The upper signature worked on the first try.  Five minutes of fiddling later, I was unable either to get the second signature to do the same, or to repeat the success with the first.

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by FrancRos31** [comment]
> Current behavior: With old key signatures (naturals cancel the previous key signature), the naturals cannot be displayed at the head of a line:
> 
> 1. Add several blank measures
> 2. Style > Accidentals > "natural in key signature: Before key signature if changing to fewer sharps or flats"
> 3. Add key signature changes that require the cancelling natural signs.
> 4. Try to get a new key signature _with the cancelling naturals_ at the head of the line.
> 
> Expected behavior: Absent any obstacle such as section breaks, the complete key signature should be displayed.

![1](https://github.com/user-attachments/assets/8d3d13af-aa2d-4bb0-93da-2a53ab38c151)
I can't replicate this issue, it works perfectly (version 4.4.2, Window 10 Pro). Maybe is a problem of Mac OS?

**Comment by RhinoHaggis** [comment]
You’ve tried at the first measure of a line?  Those all seem conspicuously in the middle.

**Comment by FrancRos31** [comment]
> 4\. Try to get a new key signature _with the cancelling naturals_ at the head of the line.

Sorry, didn't read the important "AT THE HEAD OF THE LINE", lol. You're right though, the naturals appear only in the courtesy signature. Once you disable that option, bye bye naturals.

**Comment by RhinoHaggis** [comment]
This is a little tricky and I’m hoping it might get rolled into the key sig/frame issue because secretly we  need both a natural default and also override control over the naturals. 
 I’m making copies of old scores and I think one of the reasons I never noticed this is that after a section break, usually the leading naturals are omitted altogether. So THAT default behavior is correct. However « usually » isn’t « always » and it would be a shame to do the whole (frames key signatures) over again for Old Man signatures, once this first bug gets fixed. Just a thought

**Comment by RhinoHaggis** [comment]
Ah! Maybe it’s that simple! The default behavior with « courtesy » on is correct.  Likewise with « courtesy »  turned off, after a section break:
- either there is a courtesy signature at the end of the previous section, and the key signature is cancelled, or there is no courtesy signature and the new section begins fresh, without reference to the previous section.

…and this logic is being inadvertently applied to any situation where there is a line break: softy or hard, section or not.

If that were true there’s no underlying monster, just a missing correct  behavior for simple like breaks.  After that, conforming the remaining weirdness in section break and frame behaior  would be exactly what’s already proposed for ordinary key signatures (like allowing that above-mentioned courtesy signature at the end of the previous section, and override control)

**Comment by FrancRos31** [comment]
I agree, the default behaviour with the courtesy signature is totally correct. You have to make aware the player of the key change before the system break, hence the naturals in the courtesy key signature. Do you think the program needs some override control maybe in the property panel? I remember the "Show courtesy key signature" option was in the right click menu of the courtesy key, but I'm totally fine with the two option (naturals in the key signature and show courtesy key signature) now all in the style dialog.
